Providers/Profile.swift

```diff
@@ -434,14 +439,28 @@ public class BrowserProfile: Profile {
          * Locking is managed by withSyncInputs. Make sure you take and release these
          * whenever you do anything Sync-ey.
          */
-        var syncLock = OSSpinLock()
+        var syncLock = OSSpinLock() {
+            didSet {
+                if oldValue != 0 && syncLock == 0 {
```

Comment:
Is this ever called without `oldValue != newValue`?

Seems like this could simplify to:

```
// TODO: remove this if it never occurs!
if oldValue == syncLock {
    return
}
let notification = syncLock == 0 ? ProfileDidFinishSyncingNotification : ProfileDidStartSyncingNotification
NSNotificationCenter.defaultCenter().postNotification(NSNotification(name: notification, object: nil))
```